clients/osv/src/main/kotlin/OsvService.kt

```diff
@@ -67,7 +67,9 @@ class OsvService(serverUrl: String? = null, httpClient: OkHttpClient? = null) {
             val response = client.getVulnerabilityIdsForPackages(batchRequest).execute()
             val body = response.body()
 
-            if (!response.isSuccessful || body == null) return Result.failure(IOException(response.message()))
+            if (!response.isSuccessful || body == null) {
```

Comment:
Why is `|| body == null` necessary and / or correct?

Reply:
While theoretically (i.e. on HTTP specification level) a successful response could return a null body, in the context of OSV we always expect to get a (non-nullable) `VulnerabilitiesForPackageBatchResponse`, so handling a `null` body as an error seems to be correct.

In any case, the code was like that before and is out of scope for this change.

But I'll make another change to improve error reporting via `errorBody()`.